fe/fe-core/src/main/java/com/starrocks/sql/analyzer/QueryAnalyzer.java

```diff
@@ -506,7 +506,7 @@ public Scope visitView(ViewRelation node, Scope scope) {
             for (int i = 0; i < view.getBaseSchema().size(); ++i) {
                 Column column = view.getBaseSchema().get(i);
                 Field originField = queryOutputScope.getRelationFields().getFieldByIndex(i);
-                Field field = new Field(column.getName(), column.getType(), node.getResolveTableName(),
+                Field field = new Field(column.getName(), originField.getType(), node.getResolveTableName(),
```

Comment:
What's the diff between column and originField? why name not use originField?

Reply:
A view can specify its column names optionally, if column names are absent, the output names of the queryRelation is used as the names of the view schema, so column names in view's schema are always correct. Using originField.getNames here will gives wrong names when user-specified view column names are different from output names of the queryRelation.